src/main/service.rs

```diff
@@ -75,7 +75,19 @@ impl<UID: Uid> Service<UID> {
         let mut mc = MappingContext::new()?;
         mc.add_peer_stuns(config.hard_coded_contacts.iter().cloned());
 
-        let el = common::spawn_event_loop(4, Some(&format!("{:?}", our_uid)))?;
+        let bootstrap_cache_file = config.bootstrap_cache_name.clone();
+        let el = common::spawn_event_loop(4, Some(&format!("{:?}", our_uid)), move || {
```

Comment:
Add a constant above as `const UNRESERVED_TOKEN_START_RANGE: Token = Token(4);` and use `UNRESERVED_TOKEN_START_RANGE.0` here. Best would be if rust has an incrementing C enum like:
```
enum ReservedTokens {
    BOOTSTRAP_TOKEN = 0, // Normally 0 by default anyway
    ..., // rest are auto incremented with delta of +1
    UNRESERVED,
}
```
but IDK if there's any equivalent in Rust for this.